nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/test/java/org/apache/nifi/processors/standard/TestEncryptContent.java

```diff
@@ -92,6 +96,48 @@ public void testRoundTrip() throws IOException {
         }
     }
 
+    @Test
+    public void testPgpCiphersRoundTrip() throws IOException {
+        final TestRunner testRunner = TestRunners.newTestRunner(new EncryptContent());
+        List<String> pgpAlgorithms = new ArrayList<>();
+        pgpAlgorithms.add("PGP");
+        pgpAlgorithms.add("PGP_ASCII_ARMOR");
+
+        for (String algorithm : pgpAlgorithms) {
+
+            testRunner.setProperty(EncryptContent.ENCRYPTION_ALGORITHM, algorithm);
+            testRunner.setProperty(EncryptContent.PASSWORD, "short");
+            testRunner.setProperty(EncryptContent.KEY_DERIVATION_FUNCTION, KeyDerivationFunction.NIFI_LEGACY.name());
+            // Must be allowed or short password will cause validation errors
+            testRunner.setProperty(EncryptContent.ALLOW_WEAK_CRYPTO, "allowed");
+
+            for (int i = 1; i < 14; i++) {
+                if (i != SymmetricKeyAlgorithmTags.SAFER) {
+                    testRunner.setProperty(EncryptContent.PGP_SYMMETRIC_ENCRYPTION_CIPHER, String.valueOf(i));
+                    testRunner.setProperty(EncryptContent.MODE, EncryptContent.ENCRYPT_MODE);
+
+                    testRunner.enqueue(Paths.get("src/test/resources/hello.txt"));
+                    testRunner.clearTransferState();
+                    testRunner.run();
+
+                    testRunner.assertAllFlowFilesTransferred(EncryptContent.REL_SUCCESS, 1);
+
+                    MockFlowFile flowFile = testRunner.getFlowFilesForRelationship(EncryptContent.REL_SUCCESS).get(0);
+                    testRunner.assertQueueEmpty();
+                    System.out.println(Arrays.toString(flowFile.toByteArray()));
+                    testRunner.setProperty(EncryptContent.MODE, EncryptContent.DECRYPT_MODE);
+                    testRunner.enqueue(flowFile);
+                    testRunner.clearTransferState();
+                    testRunner.run();
+                    testRunner.assertAllFlowFilesTransferred(EncryptContent.REL_SUCCESS, 1);
+
+                    flowFile = testRunner.getFlowFilesForRelationship(EncryptContent.REL_SUCCESS).get(0);
+                    flowFile.assertContentEquals(new File("src/test/resources/hello.txt"));
```

Comment:
this doesn't prove that the properties are correctly used by the processor

Reply:
Right, thanks for pointing it out. The new test case infer the cipher directly from the ciphertext to prove it was the provided one